R/utilities-simulation.R

```diff
@@ -749,3 +749,230 @@ getSimulationTree <- function(simulationOrFilePath, quantityType = "Quantity") {
 
   return(pathEnumList)
 }
+
+#' Get the steady-state values of species and state variable parameters.
+#'
+#' @details The steady-state is considered to be the last values of the molecules
+#' amounts and state variable parameters in the simulation with sufficiently long
+#' simulation time, i.e., where the rates of
+#'   the processes do not (significantly) change. The steady-state is NOT analytically
+#'   calculated or estimated in any other way than simulating for the given time.
+#'
+#'
+#' @param steadyStateTime Simulation time (minutes). Must be long enough for
+#'   system to reach a steady-state. 1000 by default. Either a single value (will
+#'   be applied for all simulations), or a vector of values specific for each
+#'   simulation. In latter case, must have equal size as `simulations`.
+#' @param quantitiesPaths List of quantity paths (molecules and/or parameters)
+#'   for which the steady-state will be simulated. If `NULL` (default), all
+#'   molecules and state variable parameters are considered. The same list is
+#'   applied for all simulations.
+#' @param simulations `Simulation` object or a list of `Simulation` objects
+#' @param ignoreIfFormula If `TRUE` (default), species and parameters with
+#'   initial values defined by a formula are not included.
+#' @param stopIfNotFound Boolean. If `TRUE` (default), an error is thrown when
+#'   results for certain species were not generated. This may happen when due to
+#'   numerical problems some values cannot be calculated, though the whole
+#'   simulation converges. Setting this argument to `FALSE` allows to ignore
+#'   such errors. Check the outputs for empty values when using this option.
```

Comment:
I don't get this: how can the values for SOME species be not available if the simulation run was successful?

Reply:
If I remember correctly, it was some observer that was `NaN` the whole time... there was a case for this.

Reply:
or, so with "species" you mean also observers? (I thought only ODE state variables were meant here).
Then this could happen indeed.